src/utils/io/writer_checksum.rs

```diff
@@ -45,24 +46,24 @@ impl<D: Digest, W: AsyncWriteExt + Unpin> AsyncWrite for AsyncWriterWithChecksum
     }
 
     fn poll_flush(
-        mut self: std::pin::Pin<&mut Self>,
+        self: std::pin::Pin<&mut Self>,
         cx: &mut std::task::Context<'_>,
     ) -> std::task::Poll<std::io::Result<()>> {
-        Pin::new(&mut self.inner).poll_flush(cx)
+        let this = self.project();
+        this.inner.poll_flush(cx)
     }
 
     fn poll_close(
-        mut self: Pin<&mut Self>,
+        self: Pin<&mut Self>,
         cx: &mut std::task::Context<'_>,
     ) -> Poll<std::io::Result<()>> {
-        Pin::new(&mut self.inner).poll_close(cx)
+        let this = self.project();
+        this.inner.poll_close(cx)
```

Comment:
nit: I'm partial to one-liners.

```suggestion
        self.project().inner.poll_close(cx)
```